src/Functions/likePatternToRegexp.h

```diff
@@ -20,53 +20,64 @@ inline String likePatternToRegexp(std::string_view pattern)
     const char * pos = pattern.data();
     const char * const end = pattern.begin() + pattern.size();
 
+    char char_slot;
     if (pos < end && *pos == '%')
-        ++pos;
+    {
+        while (++pos < end && *pos == '%')
+        {
+        }
+    }
     else
+    {
         res = "^";
-
-    while (pos < end)
+    }
+    while (pos != end)
     {
-        switch (*pos)
+        char_slot = *(pos++);
```

Comment:
> Do you think it's necessary? Because it looks much less intuitive to me. I understand it could probably be faster, but pattern strings are usually not so big. So it is reasonable to sacrifice readability in this case.

well it could be changed to have better readability instead of performance.

Reply:
@pufit done.